app/services/dual-output/dual-output.ts

```diff
@@ -360,80 +381,291 @@ export class DualOutputService extends PersistentStatefulService<IDualOutputServ
+   * Create or confirm nodes for vertical output
    */
-  confirmOrCreateVerticalNodes(sceneId: string) {
-    this.convertSceneSources(sceneId);
-    if (!this.views.hasNodeMap(sceneId) && this.state.dualOutputMode) {
-      try {
-        this.createSceneNodes(sceneId);
-      } catch (error: unknown) {
-        console.error('Error toggling Dual Output mode: ', error);
-      }
-    } else {
-      try {
-        this.confirmOrAssignSceneNodes(sceneId);
-      } catch (error: unknown) {
-        console.error('Error toggling Dual Output mode: ', error);
-      }
+  confirmOrCreateDualOutputCollection() {
+    if (this.views?.sceneNodeMaps) {
+      this.validateDualOutputCollection();
+    } else if (this.state.dualOutputMode) {
+      this.convertSingleOutputToDualOutputCollection();
     }
+    if (this.state.isLoading) this.SET_IS_LOADING(false);
+  }
+
+  validateDualOutputCollection() {
+    this.SET_IS_LOADING(true);
+    // confirm dual output scene collection
+    this.scenesService.views.scenes.forEach(scene => {
+      if (this.views.sceneNodeMaps[scene.id]) {
+        this.confirmSceneNodeMap(scene.id, this.views.sceneNodeMaps[scene.id]);
+      } else {
+        this.createVerticalNodes(scene.id);
+      }
+    });
+    this.SET_IS_LOADING(false);
   }
 
-  convertSceneSources(sceneId: string) {
-    const sceneSources = this.scenesService.views.sceneSourcesForScene(sceneId);
-    if (sceneSources.length > 0) {
-      sceneSources.forEach(scene => this.confirmOrCreateVerticalNodes(scene.sourceId));
+  convertSingleOutputToDualOutputCollection() {
+    // if in dual output mode, convert the single output collection to dual output
+    this.scenesService.views.scenes.forEach(scene => {
+      this.createVerticalNodes(scene.id);
+    });
+  }
+
+  /**
+   * Create a vertical node to partner with the vertical node
+   * @param horizontalNode - Node to copy to the vertical display
+   *
+   * @remark The horizontal node id is always the key in the scene node map.
+   * The node map entry is so that the horizontal and vertical nodes can refer to each other.
+   */
+  createVerticalNode(horizontalNode: TSceneNode, verticalNodeId?: string): TSceneNode {
+    const scene = horizontalNode.getScene();
+
+    if (horizontalNode.isFolder()) {
+      // add folder and create node map entry
+      const folder = scene.createFolder(horizontalNode.name, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+      this.sceneCollectionsService.createNodeMapEntry(scene.id, horizontalNode.id, folder.id);
+
+      this.sceneNodeHandled.next();
+      return folder;
+    } else {
+      // add item
+      const item = scene.addSource(horizontalNode.sourceId, {
+        id: verticalNodeId,
+        display: 'vertical',
+      });
+
+      // position all of the nodes in the upper left corner of the vertical display
+      // so that all of the sources are visible
+      item.setTransform({ position: { x: 0, y: 0 } });
```

Comment:
For the future, please ignore for now, would there be a way we can avoid forcing manipulating the scene item properties this way? If I remember correctly we have to workaround these on load to preserve position, visibility, etc. somewhere around the loading scene sequence, I wonder if we should at least move this code there at some point, so we'd look at one place for everything related to it, as opposed to being split.

Reply:
This was a decision that was made so that all vertical items would be available to the user when converting from a single output to a dual output collection. If we preserve the properties and don't scale the horizontal scene, some scene items could be created outside of the canvas, potentially outside of the padding boundary we set in the editor in dual output mode so that scene items couldn't be dragged out of the editor. If we scale the horizontal scene, some vertical scene items could be too small to interact with. Repositioning the sources was not actually a decision I made. I'm open to another solution (such as partial scaling) but this would be a product decision.